virtcontainers/qemu.go

```diff
@@ -409,6 +409,10 @@ func (q *qemu) buildDevices(initrdPath string) ([]govmmQemu.Device, *govmmQemu.I
 		}
 	}
 
+	if q.config.IOMMU {
+		devices = q.arch.appendIOMMU(devices)
```

Comment:
I'm curious what the semantics is for the returned value. Is `appendIOMMU` supposed to prune the list or add more devices? Is that the reason for returning a device list? All the code examples I looked at returned their input.

Reply:
Yes. It's the same semantics as [golang's slice appending](https://golang.org/pkg/builtin/#append)